### PR TITLE
Add frontend API for datasetOwnedBy

### PR DESCRIPTION
**Pull request by alyiwang**
- Add frontend API for datasetOwnedBy, GET paged datasets owned by a user. 
- Enforce a minimum of 2 confirmed users before making changes on dataset owners page.
- Fix minor issues on frontend Dashboard APIs.

### CODE REVIEW

File: web/app/dao/DatasetsDAO.java [hunk 1]
```diff
@@ -1733,96 +1785,69 @@ public int getBatchSize()
...
 	}
 
-	public static boolean updateDatasetOwners(int datasetId, Map<String, String[]> ownersMap, String user)
-	{
-		boolean result = true;
-		if ((ownersMap == null) || ownersMap.size() == 0)
-		{
-			return false;
+	public static int updateDatasetOwners(int datasetId, Map<String, String[]> ownersMap, String user) {
+		if ((ownersMap == null) || ownersMap.size() == 0 || !ownersMap.containsKey("owners")
```
Comment (theseyi) [inline web/app/dao/DatasetsDAO.java]: Maybe add a comment here explaining the intention behind the condition.

File: web/app/controllers/api/v1/Dataset.java [hunk 2]
```diff
@@ -233,33 +233,30 @@ public static Result getDatasetImpactAnalysisByID(int id)
...
-            {
+        if (StringUtils.isNotBlank(username)) {
+            int returnCode = DatasetsDAO.updateDatasetOwners(id, params, username);
+
+            if (returnCode == 0) {
                 result.put("status", "success");
-            }
-            else
-            {
+            } else if (returnCode == 2) {
```
Comment (mars-lan) [inline web/app/controllers/api/v1/Dataset.java]: Define enums instead of integer for different cases?

File: web/app/dao/DatasetsDAO.java (hunk 1 above)
Comment (alyiwang) [inline web/app/dao/DatasetsDAO.java]: Comments added

File: web/app/controllers/api/v1/Dataset.java (hunk 2 above)
Comment (alyiwang) [inline web/app/controllers/api/v1/Dataset.java]: Updated with Return Code enum.

**Pull request merged**